google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudLibraryDependencyWriter.java

```diff
@@ -121,23 +130,81 @@ protected void run(@NotNull Result result) throws Throwable {
-                    .ping();
-              });
+          newMavenIds.forEach(mavenId -> writeNewMavenDependency(model, mavenId, bomVersion));
+
+          if (bomVersion != null) {
+            addBomToMavenModule(model, bomVersion);
+          }
+
           notifyAddedDependencies(newMavenIds, project);
         }
       }
     }.execute();
   }
 
+  /**
+   * Adds the google-cloud-java BOM dependency to the module's dependencyManagement section.
+   *
+   * <p>If the BOM already exists, then this will update the version of the BOM to match the
+   * supplied version.
+   */
+  private static void addBomToMavenModule(MavenDomProjectModel model, String bomVersion) {
+    MavenDomDependencies mavenDomDependencies = model.getDependencyManagement().getDependencies();
+    Optional<MavenDomDependency> bomDependencyOptional =
+        mavenDomDependencies
+            .getDependencies()
+            .stream()
+            .filter(
+                mdd ->
+                    CloudApiMavenService.GOOGLE_CLOUD_JAVA_BOM_ARTIFACT.equals(
+                        mdd.getArtifactId().getStringValue()))
+            .findFirst();
+
+    if (!bomDependencyOptional.isPresent()) {
+      writeNewBom(model, bomVersion);
+    } else {
+      // update the version
+      bomDependencyOptional.get().getVersion().setStringValue(bomVersion);
+    }
+  }
+
+  /**
+   * Writes the google-cloud-java dependency to the users pom.xml.
+   *
+   * <p>If a BOM version is supplied, the dependency will not be written with a version so that the
+   * BOM can manage the version.
+   */
+  private static void writeNewMavenDependency(
+      MavenDomProjectModel model, MavenId mavenId, @Nullable String bomVersion) {
+    MavenDomDependency dependency = MavenDomUtil.createDomDependency(model, /* editor= */ null);
+    dependency.getGroupId().setStringValue(mavenId.getGroupId());
+    dependency.getArtifactId().setStringValue(mavenId.getArtifactId());
+
+    // Only write the version to the dependency if there is no BOM
+    if (bomVersion == null) {
+      dependency.getVersion().setStringValue(mavenId.getVersion());
+    }
+
+    UsageTrackerProvider.getInstance()
+        .trackEvent(GctTracking.CLIENT_LIBRARY_ADD_LIBRARY)
+        .addMetadata(GctTracking.METADATA_BUILD_SYSTEM_KEY, GctTracking.METADATA_BUILD_SYSTEM_MAVEN)
+        .addMetadata(GctTracking.METADATA_LABEL_KEY, mavenId.getDisplayString())
+        .ping();
+  }
+
+  /** Writes the google-cloud-java BOM to the user's pom.xml. */
+  private static void writeNewBom(MavenDomProjectModel model, String bomVersion) {
+    MavenDomDependency bomDependency =
+        model.getDependencyManagement().getDependencies().addDependency();
+    bomDependency.getGroupId().setStringValue(CloudApiMavenService.GOOGLE_CLOUD_JAVA_BOM_GROUP);
+    bomDependency
+        .getArtifactId()
+        .setStringValue(CloudApiMavenService.GOOGLE_CLOUD_JAVA_BOM_ARTIFACT);
+    bomDependency.getVersion().setStringValue(bomVersion);
```

Comment:
Could we also benefit from collecting usage of specific BOM versions people use, or maybe just latest/non-latest flag?

Reply:
absolutely. I intended to add in metrics for the entire BOM effort in a separate PR.